File: 30DaysOfCode/day08/dictionariesAndMaps.js
function processData(input) {
    inputArray = input.split("\n");
    
    var N = parseInt(inputArray.shift());
    var phonebook = {};
    
    for (var i = 0; i < N; i++) {
        var S = inputArray.shift();
        SArray = S.split(" ");
        
        phonebook[SArray[0]] = SArray[1];
    }
    
    while (inputArray.length > 0) {
        var S = inputArray.shift();
        
        if (!(S in phonebook)) {
            console.log("Not found");
        }
        else {
            console.log(S + "=" + phonebook[S]);
        }
    }
} 

process.stdin.resume();
process.stdin.setEncoding("ascii");

_input = "";
process.stdin.on("data", function (input) {
    _input += input;
});

process.stdin.on("end", function () {
   processData(_input);
});

File: 30DaysOfCode/day08/dictionariesAndMaps.py
import sys

N = int(raw_input())
phonebook = {}

for i in range(0, N):
    S = str(raw_input()).split(" ")
    phonebook[S[0]] = S[1]
    
lines = sys.stdin.readlines()
for line in lines:
    if line.strip() in phonebook:
        sys.stdout.write(line.strip())
        sys.stdout.write("=")
        sys.stdout.write(phonebook[line.strip()])
        sys.stdout.write("\n")
    else:
        print "Not found"
